timemachine/fe/bar.py

```diff
@@ -145,22 +145,21 @@ def bootstrap_bar(w_F, w_R, n_bootstrap=1000, timeout=10) -> Tuple[float, NDArra
         w_F_sample = rng.choice(w_F, size=(n_F,), replace=True)
         w_R_sample = rng.choice(w_R, size=(n_R,), replace=True)
 
-        bar_result = pymbar.BAR(
+        bar_result = pymbar.bar(
             w_F=w_F_sample,
             w_R=w_R_sample,
-            DeltaF=full_bar_result,  # warm start
+            DeltaF=full_bar_result["Delta_f"],  # warm start
             compute_uncertainty=False,
             relative_tolerance=1e-6,  # reduce cost
         )
 
-        bootstrap_samples.append(bar_result)
+        bootstrap_samples.append(bar_result["Delta_f"])
 
-    return full_bar_result, np.array(bootstrap_samples)
+    return full_bar_result["Delta_f"], np.array(bootstrap_samples)
 
 
 def bar_with_bootstrapped_uncertainty(w_F, w_R, n_bootstrap=1000, timeout=10) -> Tuple[float, float]:
-    """Drop-in replacement for pymbar.BAR(w_F, w_R) -> (df, ddf)
-    where first return is forwarded from pymbar.BAR but second return is computed by bootstrapping"""
+    """Returns BAR estimate from pymbar.bar and error estimate computed by bootstrapping."""
```

Comment:
After this upgrade, we can probably remove the bootstrapping code (added in https://github.com/proteneer/timemachine/pull/829 ) and use pymbar4's implementation.

Reply:
Yes, I'll do that in a follow-up -- thanks for paving the way!